pkg/cloudprovider/providers/azure/azure_vmss_cache.go

```diff
@@ -99,7 +99,7 @@ func (ss *scaleSet) newNodeNameToScaleSetMappingCache() (*timedCache, error) {
 
 			for _, vm := range vms {
 				if vm.OsProfile == nil || vm.OsProfile.ComputerName == nil {
-					glog.Warningf("failed to get computerName for vmssVM (%q)", vm.Name)
```

Comment:
it should be `v.Name`, pls revert this code change

Reply:
`vm.Name` is a `*string` (pointer to string), not a string, and `%q` will not print it. I thought about using `*vm.Name` but I didn't want to turn what used to be a buggy warning into an actual crash (if `vm.Name` was nil). 

Should I use `*vm.Name`? Is that guaranteed to be non-nil?

Reply:
@rsc Then I would suggest keep the original code:
```
glog.Warningf("failed to get computerName for vmssVM (%q)", vm.Name)
```

This is `fix printf format errors` PR, not bug fix.

Reply:
@andyzhangx Yes, this is a "fix printf format errors" PR. 

I am editing that line ***because it has a printf format error***. 

Compiled using Go 1.11, the pkg/cloudprovider/providers/azure test fails (because test implies vet for printf format errors, and go vet has gotten better at detecting printf wrappers like glog.Warningf):

```
# k8s.io/kubernetes/pkg/cloudprovider/providers/azure
pkg/cloudprovider/providers/azure/azure_client.go:848: Verbose.Infof format %q reads arg #2, but call has 1 arg
pkg/cloudprovider/providers/azure/azure_client.go:850: Verbose.Infof format %q reads arg #2, but call has 1 arg
pkg/cloudprovider/providers/azure/azure_client.go:879: Verbose.Infof format %q has arg VMInstanceIDs of wrong type k8s.io/kubernetes/vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2017-12-01/compute.VirtualMachineScaleSetVMInstanceRequiredIDs
pkg/cloudprovider/providers/azure/azure_client.go:881: Verbose.Infof format %q has arg VMInstanceIDs of wrong type k8s.io/kubernetes/vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2017-12-01/compute.VirtualMachineScaleSetVMInstanceRequiredIDs
pkg/cloudprovider/providers/azure/azure_loadbalancer.go:265: Verbose.Infof format %s has arg isInternal of wrong type bool
pkg/cloudprovider/providers/azure/azure_loadbalancer.go:845: Verbose.Infof format %q reads arg #3, but call has 2 args
pkg/cloudprovider/providers/azure/azure_vmss.go:478: Verbose.Info call has possible formatting directive %q
pkg/cloudprovider/providers/azure/azure_vmss.go:848: Verbose.Infof format %q reads arg #1, but call has 0 args
pkg/cloudprovider/providers/azure/azure_vmss_cache.go:102: Warningf format %q has arg vm.Name of wrong type *string
```

This PR fixes all of those errors, and more in other directories.

The line we're commenting on is causing the final reported error. Leaving the current line unmodified will mean that this test starts failing when Kubernetes updates to Go 1.11. 

I am happy to use any line you suggest provided it does NOT have a printf format error.

Reply:
ok, then let's keep this change, there is not good way since we cannot ensure `vm.Name` is not nil

Reply:
Thanks!